ddl/column.go

```diff
@@ -0,0 +1,420 @@
+		newColumns := make([]*model.ColumnInfo, 0, len(tblInfo.Columns))
+		for _, col := range tblInfo.Columns {
+			if col.Name.L != colName.L {
+				newColumns = append(newColumns, col)
+			}
+		}
+		tblInfo.Columns = newColumns
+		if err = t.UpdateTable(schemaID, tblInfo); err != nil {
+			return errors.Trace(err)
+		}
+
+		// finish this job
+		job.SchemaState = model.StateNone
+		job.State = model.JobDone
+		return nil
+	default:
+		return errors.Errorf("invalid table state %v", tblInfo.State)
+	}
+}
+
+// How to backfill column data in reorganization state?
+//  1. Generate a snapshot with special version.
+//  2. Traverse the snapshot, get every row in the table.
+//  3. For one row, if the row has been already deleted, skip to next row.
+//  4. If not deleted, check whether column data has existed, if existed, skip to next row.
+//  5. If column data doesn't exist, backfill the column with default value and then continue to handle next row.
+func (d *ddl) backfillColumn(t table.Table, columnInfo *model.ColumnInfo, reorgInfo *reorgInfo) error {
+	seekHandle := reorgInfo.Handle
+	version := reorgInfo.SnapshotVer
+
+	for {
+		handles, err := d.getSnapshotRows(t, version, seekHandle)
+		if err != nil {
+			return errors.Trace(err)
+		} else if len(handles) == 0 {
+			return errors.Trace(reorgInfo.RemoveHandle())
+		}
+
+		seekHandle = handles[len(handles)-1] + 1
+		// TODO: save seekHandle in reorganization job, so we can resume this job later from this handle.
+
+		err = d.backfillColumnData(t, columnInfo, handles, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+}
+
+func (d *ddl) backfillColumnData(t table.Table, columnInfo *model.ColumnInfo, handles []int64, reorgInfo *reorgInfo) error {
+	for _, handle := range handles {
+		log.Info("backfill column...", handle)
+
+		err := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
+			// First check if row exists.
+			exist, err := checkRowExist(txn, t, handle)
+			if err != nil {
+				return errors.Trace(err)
+			} else if !exist {
+				// If row doesn't exist, skip it.
+				return nil
+			}
+
+			backfillKey := t.RecordKey(handle, &column.Col{ColumnInfo: *columnInfo})
+			backfillValue, err := txn.Get(backfillKey)
+			if err != nil && !kv.IsErrNotFound(err) {
+				return errors.Trace(err)
+			}
+			if backfillValue != nil {
+				return nil
+			}
+
+			value, _, err := tables.GetColDefaultValue(nil, columnInfo)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			// must convert to the column field type.
+			v, err := types.Convert(value, &columnInfo.FieldType)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			err = lockRow(txn, t, handle)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			err = t.SetColValue(txn, backfillKey, v)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			return errors.Trace(reorgInfo.UpdateHandle(txn, handle))
+		})
+
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+
```

Comment:
Filling one each time is just for safe, and the retry cost is tiny. 
But later we may fill multi columns if possible, and it is not hard to support.